protocol-definitions/Client.yaml

```diff
@@ -774,4 +774,112 @@ methods:
       retryable: true
       partitionIdentifier: -1
     response:
-      {}
\ No newline at end of file
+      {}
+  - id: 17
+    name: addMigrationListener
+    since: 2.1
+    doc: |
+      Adds a Migration listener to the cluster.
+    request:
+      retryable: false
+      partitionIdentifier: -1
+      params:
+        - name: localOnly
+          type: boolean
+          nullable: false
+          since: 2.1
+          doc: |
+            if true only node that has the migration sends the request, if false
+            sends all partition lost events.
+    response:
+      params:
+        - name: response
+          type: UUID
+          nullable: false
+          since: 2.1
+          doc: |
+            The listener registration id.
+    events:
+      - name: Migration
+        since: 2.1
+        params:
+          - name: migrationState
+            type: MigrationState
+            nullable: false
+            since: 2.1
+            doc: |
+              Migration state.
+          - name: partitionId
```

Comment:
`partition id` name is confusing. Can you name it as `event type`? 
Can you also list possible values on the doc part?  We generally use positive values. 
MIGRATION_STARTED 0
MIGRATION_COMLETED 1